### ISSUE TITLE
asp.net core 5.0 opentelemetry is not generating traces for http request coming in 

### ISSUE THREAD

**Issue by Dheerajcode2016** [open-telemetry/opentelemetry-dotnet#2050]
# Bug Report

Packages used in project 
``` 
  <ItemGroup>
    <PackageReference Include="Microsoft.VisualStudio.Azure.Containers.Tools.Targets" Version="1.10.13" />
    <PackageReference Include="OpenTelemetry.Exporter.Console" Version="1.1.0-beta3" />
    <PackageReference Include="OpenTelemetry.Exporter.Jaeger" Version="1.1.0-beta3" />
    <PackageReference Include="OpenTelemetry.Extensions.Hosting" Version="1.0.0-rc4" />
    <PackageReference Include="OpenTelemetry.Instrumentation.AspNetCore" Version="1.0.0-rc4" />
    <PackageReference Include="OpenTelemetry.Instrumentation.Http" Version="1.0.0-rc4" />
    <PackageReference Include="Swashbuckle.AspNetCore" Version="6.1.4" />
  </ItemGroup>
```

*

Runtime 
```
  <PropertyGroup>
    <TargetFramework>net5.0</TargetFramework>
    <UserSecretsId>30a3fffe-4ad1-4e93-8ab2-66e18ca23da5</UserSecretsId>
    <DockerDefaultTargetOS>Linux</DockerDefaultTargetOS>
    <DockerComposeProjectPath>..\docker-compose.dcproj</DockerComposeProjectPath>
  </PropertyGroup>
```
*

## Symptom

when i am sending request from either swagger or putting url in browser then no traces are generated neither on console nor in jeager

i should traces in jeager and console.

but i don't see any trace generated

Startup.cs 

```
  public void ConfigureServices(IServiceCollection services)
    {
      services.AddControllers();
      services.AddSwaggerGen(c =>
      {
        c.SwaggerDoc("v1", new OpenApiInfo { Title = "OTel.POC.Web", Version = "v1" });
      });
      services.AddOpenTelemetryTracing((builder) =>
          builder
            .SetResourceBuilder(ResourceBuilder.CreateDefault().AddService("OTel.POC.Web"))
            .AddHttpClientInstrumentation()
            .AddAspNetCoreInstrumentation(option=> { option.RecordException = true; })
            .AddConsoleExporter()
            .AddJaegerExporter(options =>
            {
              options.AgentHost = "localhost";
              options.AgentPort = 6831;
            })
           
          );
    }
```

program.cs
```
public class Program
  {
    public static void Main(string[] args)
    {
      Activity.DefaultIdFormat = ActivityIdFormat.W3C;
      var listener = new ActivityListener
      {
        ShouldListenTo = _ => true,
        ActivityStopped = activity =>
        {
          foreach (var (key, value) in activity.Baggage)
          {
            activity.AddTag(key, value);
          }
        }

      };
      ActivitySource.AddActivityListener(listener);
      CreateHostBuilder(args).Build().Run();
    }

    public static IHostBuilder CreateHostBuilder(string[] args) =>
        Host.CreateDefaultBuilder(args)
            .ConfigureWebHostDefaults(webBuilder =>
            {
              webBuilder.UseStartup<Startup>();
            })
            .ConfigureLogging((context, builder) =>
            {

              builder.AddConsole();

              builder.AddOpenTelemetry(options =>
              {
                options.IncludeScopes = true;
                options.ParseStateValues = true;
                options.IncludeFormattedMessage = true;

                options.AddConsoleExporter();
              });
            });
  }
}
```

Repo link 
https://github.com/Dheerajcode2016/OpenTelemetry-dotnet

**Comment by utpilla** [comment]
@Dheerajcode2016 There is a breaking change with the 1.1.0-beta3 version of the SDK. There is an issue created to track this #2051.

As a temporary fix, could you please downgrade the Console and Jaeger exporter packages to version 1.1.0-beta2 to unblock yourself?

**Comment by Dheerajmentor** [comment]
Thanks @utpilla , I will try and respond back

**Comment by Dheerajmentor** [comment]
@utpilla 
I tried as you suggested but still no traces are to seen in jeager.  :-( 
Please refer my repo , i have committed the changes

**Comment by Dheerajcode2016** [comment]
Please refer my commit 
https://github.com/Dheerajcode2016/OpenTelemetry-dotnet/commit/ccd6de614d3c00610d86f46fc8e1677be12bdb48

**Comment by cijothomas** [comment]
> Please refer my commit
> [Dheerajcode2016/OpenTelemetry-dotnet@ccd6de6](https://github.com/Dheerajcode2016/OpenTelemetry-dotnet/commit/ccd6de614d3c00610d86f46fc8e1677be12bdb48)
> i tried but not working

<PackageReference Include="OpenTelemetry" Version="1.1.0-beta3" /> -- Please use beta2 of this package too.

**Comment by Vladimir-Gordiychuk** [comment]
Had the same problem using OpenTelemetry.Extensions.Hosting 1.4.0-rc.1.
Downgrade to OpenTelemetry.Extensions.Hosting 1.0.0-rc9.9 seems to solve the problem.

**Comment by Kielek** [comment]
@Vladimir-Gordiychuk , any chance that is it related to: https://github.com/open-telemetry/opentelemetry-dotnet/discussions/4051 ?

**Comment by Vladimir-Gordiychuk** [comment]
> @Vladimir-Gordiychuk , any chance that is it related to: #4051 ?

Thank you Piotr (@Kielek) for mentioning `.StartWithHost()` method (yeah, I've been in that trap already), but no, this time it wasn't the reason for the issue I had.

If I'll come up with new info - I'll come back to leave feedback.

**Comment by cijothomas** [comment]
Closing old issues, as the whole xt.Hosting packages have changed and now stable.
Please open a new issue, if there are issues needing help. Thanks.
